src/Raven.Client/Documents/InMemoryDocumentSessionOperations.cs

```diff
@@ -49,6 +50,27 @@ public abstract class InMemoryDocumentSessionOperations : IDisposable
         private BatchOptions _saveChangesOptions;
 
         /// <summary>
+        /// Store events
+        /// </summary>
+        public delegate void BeforeStoreDelegate(string id, object entityInstance, IDictionary<string, string> metadata, InMemoryDocumentSessionOperations session);
```

Comment:
By convention, the sender (the session) should be the first item in all those events